junit/src/main/java/org/kaazing/k3po/junit/rules/Latch.java

```diff
@@ -28,6 +29,8 @@
     private final CountDownLatch startable;
     private final CountDownLatch finished;
     private volatile Thread testThread;
+    
+    private final AtomicBoolean exceptionOccurred = new AtomicBoolean(false); 
```

Comment:
One more thing that I see, there is no sync on the controller that sends commands to the k3po driver. In case of `notifyBarrier` and `awaitBarrier`, these are sent from the main thread, but the rest of the communication is done on the second thread. I think this might cause problems. Maybe we need a synchronize or something.